homework/homework5/node.py

```diff
@@ -0,0 +1,173 @@
+from copy import deepcopy
+from random import random
+from typing import Tuple, Optional, Any
+from json import dumps, JSONEncoder, JSONDecoder
+
+
+class Node:
```

Comment:
Sorry for my nerdy behaviour, but I would like to see docstrings in the code at least for key structures - Node, NodeEncoder, NodeDecoder, Treap